src/main/java/io/github/moulberry/notenoughupdates/overlays/EquipmentOverlay.java

```diff
@@ -205,7 +205,28 @@ public void renderGuis(GuiInventory inventory) {
 		int mouseX = Mouse.getX() * width / Minecraft.getMinecraft().displayWidth;
 		int mouseY = height - Mouse.getY() * height / Minecraft.getMinecraft().displayHeight - 1;
 
+		// Draw Backgrounds
 		GL11.glColor4f(1F, 1F, 1F, 1F);
+		AccessorGuiContainer container = ((AccessorGuiContainer) inventory);
+		final int overlayLeft = container.getGuiLeft() - ARMOR_OVERLAY_OVERHAND_WIDTH;
+		final int overlayTop = container.getGuiTop();
+		if (shouldRenderArmorHud) {
+			ResourceLocation equipmentTexture = getCustomEquipmentTexture(shouldRenderPets);
+			Minecraft.getMinecraft().getTextureManager().bindTexture(equipmentTexture);
+
+			Utils.drawTexturedRect(overlayLeft, overlayTop, ARMOR_OVERLAY_WIDTH, ARMOR_OVERLAY_HEIGHT, GL11.GL_NEAREST);
+		}
+
+		if (shouldRenderPets) {
+			ResourceLocation customPetTexture = getCustomPetTexture(shouldRenderArmorHud);
+			Minecraft.getMinecraft().getTextureManager().bindTexture(customPetTexture);
+			GlStateManager.color(1, 1, 1, 1);
+
+			Utils.drawTexturedRect(overlayLeft, overlayTop + PET_OVERLAY_OFFSET_Y, PET_OVERLAY_WIDTH, PET_OVERLAY_HEIGHT, GL11.GL_NEAREST);
+		}
+		GlStateManager.bindTexture(0);
+
+		// Draw foregrounds
```

Comment:
I don't see the point behind moving the background rendering in here. If it really needs to be this way, maybe move it out into a standalone method.

Reply:
Thats required for the background of the pet-hud to not occlude parts of the equipment-hub. This system should probably be changed to use 3 textures: one for equipment only, one for pets only and one thats both together. (rn it's 3/4 depending on the GUI style...)

For now, moved it to another method.